flamingo/common.h

```diff
@@ -50,50 +50,303 @@ static inline int find_static_members_in_class(flamingo_t* flamingo, flamingo_sc
 
 // Environment prototypes.
 
+/**
+ * Allocate a new environment.
+ *
+ * @return The new environment.
+ */
 static inline flamingo_env_t* env_alloc(void);
+
+/**
+ * Free an environment.
+ *
+ * This frees the environment and decrements the reference count of all scopes in its stack.
+ *
+ * @param env The environment to free.
+ */
 static inline void env_free(flamingo_env_t* env);
+
+/**
+ * Create a closure over an environment.
+ *
+ * This creates a new environment that shares the scopes of the given environment.
+ * The reference counts of the scopes are incremented.
+ *
+ * @param env The environment to close over.
+ * @return The new environment.
+ */
 static inline flamingo_env_t* env_close_over(flamingo_env_t* env);
+
+/**
+ * Get the parent scope of the current scope in the environment.
+ *
+ * @param env The environment.
+ * @return The parent scope, or NULL if there is no parent scope (stack size < 2).
+ */
 static inline flamingo_scope_t* env_parent_scope(flamingo_env_t* env);
+
+/**
+ * Get the current scope in the environment.
+ *
+ * @param env The environment.
+ * @return The current scope.
+ */
 static inline flamingo_scope_t* env_cur_scope(flamingo_env_t* env);
+
+/**
+ * Attach a scope to the environment's stack.
+ *
+ * This pushes the scope onto the stack.
+ *
+ * @param env The environment.
+ * @param scope The scope to attach.
+ */
 static inline void env_gently_attach_scope(flamingo_env_t* env, flamingo_scope_t* scope);
+
+/**
+ * Detach the current scope from the environment's stack.
+ *
+ * This pops the scope from the stack and returns it.
```

Comment:
maybe we could specify that this doesn't free or change the refcount of the scope, as opposed to `env_pop_scope`? maybe also an example of where this would be useful, e.g. when we create an instance of a class we want to keep its scope as its instance.

something similar for `env_gently_attach_scope` but that's less important imo